### PR TITLE
Search ror 1764

### PR DESCRIPTION
**Pull request by prakhyatox**
Ticket: https://github.com/FAIRsharing/fairsharing.github.io/issues/1764

WDID:
Fetching the organisations from ROR API ``` https://api.ror.org/organizations?query=TEXT_TO_QUERY``` and allowing user to select the organisation from the drop down list

Demo Video:
[Search-ROR.webm](https://user-images.githubusercontent.com/109139583/186625137-03ac0fc8-7d89-465e-9221-ff3d3fcc4cd6.webm)

### CODE REVIEW

File: src/components/Editor/Organisations/LinkOverlay.vue [hunk 1]
```diff
@@ -100,6 +100,28 @@
...
                 </v-card-text>
+                <v-card-text
+                  v-if="!validName"
+                  class="mb-0 pb-0"
+                >
+                  <v-alert
+                    type="error"
+                    class="mb-0"
+                  >
+                    Enter Valid Organisation Name
```
Comment (knirirr) [inline src/components/Editor/Organisations/LinkOverlay.vue]:
```suggestion
                    No matching organisation found.
```

Comment (prakhyatox) [reply src/components/Editor/Organisations/LinkOverlay.vue]: Fixed

File: src/components/Editor/Organisations/LinkOverlay.vue [hunk 2]
```diff
@@ -112,11 +134,26 @@
                           cols="12"
                           class="pb-0"
                         >
+                          <!-- Enter name of organization -->
```
Comment (knirirr) [inline src/components/Editor/Organisations/LinkOverlay.vue]:
```suggestion
                          <!-- Enter name of organisation -->
```

File: src/components/Editor/Organisations/LinkOverlay.vue [hunk 3]
```diff
@@ -223,7 +271,22 @@
                   </v-form>
                 </v-card-text>
                 <v-card-actions>
+                  <v-btn
+                    class="success"
+                    :disabled="importROR"
+                    :loading="menus.newOrganisation.loading"
+                    @click="getOrganizations()"
```
Comment (knirirr) [inline src/components/Editor/Organisations/LinkOverlay.vue]:
```suggestion
                    @click="getOrganisations()"
```

File: src/components/Editor/Organisations/LinkOverlay.vue [hunk 4]
```diff
@@ -550,7 +633,78 @@
...
-        }
+        },
+
+        addNewOrganisation() {
+          this.menus.show = 'organisation'
+          this.enterName = true
+          this.importROR = false
+        },
+
+        async getOrganizations() {
```
Comment (knirirr) [inline src/components/Editor/Organisations/LinkOverlay.vue]:
```suggestion
        async getOrganisations() {
```

File: src/components/Editor/Organisations/LinkOverlay.vue [hunk 5]
```diff
@@ -550,7 +633,78 @@
...
+          this.menus.newOrganisation.loading = true;
+          let orgName = (' ' + this.menus.newOrganisation.data.name).slice(1).trim()// make a copy of the string and trim it
+          let data = await pubClient.getROROrganization(orgName);
```
Comment (knirirr) [inline src/components/Editor/Organisations/LinkOverlay.vue]:
```suggestion
          let data = await pubClient.getROROrganisation(orgName);
```

File: src/lib/Client/ExternalClients.js [hunk 6]
```diff
@@ -18,6 +18,7 @@ class ExternalRESTClients {
         this.pmidBaseURL = "https://eutils.ncbi.nlm.nih.gov/entrez/eutils/esummary.fcgi?db=pubmed&retmode=json&id=";
         this.tessBaseURL = "https://tess.elixir-europe.org/materials.json?q=";
         this.orcidBaseURL = "https://pub.orcid.org/v2.0/";
+        this.rorOrganizationsBaseURL = "https://api.ror.org/organizations?query=";
```
Comment (knirirr) [inline src/lib/Client/ExternalClients.js]:
```suggestion
        this.rorOrganisationsBaseURL = "https://api.ror.org/organizations?query=";
```

File: src/lib/Client/ExternalClients.js [hunk 7]
```diff
@@ -62,6 +63,17 @@ class ExternalRESTClients {
         return response.data;
     }
 
+    async getROROrganization(organization){
```
Comment (knirirr) [inline src/lib/Client/ExternalClients.js]:
```suggestion
    async getROROrganisation(organisation){
```

File: src/lib/Client/ExternalClients.js [hunk 8]
```diff
@@ -62,6 +63,17 @@ class ExternalRESTClients {
...
+        let localHeaders = this.headers;
+        localHeaders['Accept'] = 'application/json';
+        const request = {
+            url: this.rorOrganizationsBaseURL + organization,
```
Comment (knirirr) [inline src/lib/Client/ExternalClients.js]:
```suggestion
            url: this.rorOrganisationsBaseURL + organisation,
```

File: tests/unit/components/Editor/Organisations/LinkOverlay.spec.js [hunk 9]
```diff
@@ -173,4 +187,106 @@ describe("Edit -> LinkOverlay.vue", function() {
...
+        wrapper.vm.menus.show = '';
+        wrapper.vm.enterName = false
+        wrapper.vm.importROR = true
+        wrapper.vm.addNewOrganisation()
+        expect(wrapper.vm.menus.show).toBe('organisation')
+        expect(wrapper.vm.enterName).toBe(true)
+        expect(wrapper.vm.importROR).toBe(false)
+    });
+
+    it("can getOrganizations List", async() => {
```
Comment (knirirr) [inline tests/unit/components/Editor/Organisations/LinkOverlay.spec.js]:
```suggestion
    it("can getOrganisations List", async() => {
```

File: tests/unit/components/Editor/Organisations/LinkOverlay.spec.js [hunk 10]
```diff
@@ -173,4 +187,106 @@ describe("Edit -> LinkOverlay.vue", function() {
...
+                },
+                "links":["http://www.obsmedical.com/"],
+                "name":"OBS Medical (United Kingdom)",
+                "types":["Company"],
+            }]
+        wrapper.vm.enterName = true
+        wrapper.vm.importROR = false
+        wrapper.vm.validName = true
+        wrapper.vm.menus.newOrganisation.data.name = 'Oxford'
+        await wrapper.vm.getOrganizations();
```
Comment (knirirr) [inline tests/unit/components/Editor/Organisations/LinkOverlay.spec.js]:
```suggestion
        await wrapper.vm.getOrganisations();
```

File: tests/unit/components/Editor/Organisations/LinkOverlay.spec.js [hunk 11]
```diff
@@ -173,4 +187,106 @@ describe("Edit -> LinkOverlay.vue", function() {
...
+        expect(wrapper.vm.enterName).toBe(false)
+        expect(wrapper.vm.importROR).toBe(true)
+        expect(wrapper.vm.validName).toBe(true)
+        expect(wrapper.vm.organisationsList).toStrictEqual(expectedOrganisation);
+        expect(wrapper.vm.organisationsNameList).toStrictEqual(["OBS Medical (United Kingdom)"])
+        fetchStub.restore();
+
+    });
+
+    it("cannot getOrganizations List", async() => {
```
Comment (knirirr) [inline tests/unit/components/Editor/Organisations/LinkOverlay.spec.js]:
```suggestion
    it("cannot getOrganisations List", async() => {
```

File: tests/unit/components/Editor/Organisations/LinkOverlay.spec.js [hunk 12]
```diff
@@ -173,4 +187,106 @@ describe("Edit -> LinkOverlay.vue", function() {
...
+        const fetchOrganisation = {}
+        fetchStub = sinon.stub(ExternalClient.prototype, "executeQuery");
+        fetchStub.returns({data:fetchOrganisation});
+        const expectedOrganisation = []
+        wrapper.vm.enterName = true
+        wrapper.vm.importROR = false
+        wrapper.vm.validName = true
+        wrapper.vm.menus.newOrganisation.data.name = 'abc'
+        await wrapper.vm.getOrganizations();
```
Comment (knirirr) [inline tests/unit/components/Editor/Organisations/LinkOverlay.spec.js]:
```suggestion
        await wrapper.vm.getOrganisations();
```

**Review by knirirr — APPROVED** [review]
I've changed the spellings for consistency with the British variant used elsewhere in the application (and on the server).

File: src/components/Editor/Organisations/LinkOverlay.vue (hunk 2 above)
Comment (prakhyatox) [reply src/components/Editor/Organisations/LinkOverlay.vue]: Fixed

File: src/components/Editor/Organisations/LinkOverlay.vue (hunk 3 above)
Comment (prakhyatox) [reply src/components/Editor/Organisations/LinkOverlay.vue]: Fixed

File: src/lib/Client/ExternalClients.js (hunk 6 above)
Comment (prakhyatox) [reply src/lib/Client/ExternalClients.js]: Fixed

File: src/lib/Client/ExternalClients.js (hunk 8 above)
Comment (prakhyatox) [reply src/lib/Client/ExternalClients.js]: Fixed

File: tests/unit/components/Editor/Organisations/LinkOverlay.spec.js (hunk 10 above)
Comment (prakhyatox) [reply tests/unit/components/Editor/Organisations/LinkOverlay.spec.js]: Fixed

File: tests/unit/components/Editor/Organisations/LinkOverlay.spec.js (hunk 11 above)
Comment (prakhyatox) [reply tests/unit/components/Editor/Organisations/LinkOverlay.spec.js]: Fixed

File: tests/unit/components/Editor/Organisations/LinkOverlay.spec.js (hunk 12 above)
Comment (prakhyatox) [reply tests/unit/components/Editor/Organisations/LinkOverlay.spec.js]: Fixed

File: tests/unit/components/Editor/Organisations/LinkOverlay.spec.js (hunk 9 above)
Comment (prakhyatox) [reply tests/unit/components/Editor/Organisations/LinkOverlay.spec.js]: Fixed

File: src/lib/Client/ExternalClients.js (hunk 7 above)
Comment (prakhyatox) [reply src/lib/Client/ExternalClients.js]: Fixed

File: src/components/Editor/Organisations/LinkOverlay.vue (hunk 5 above)
Comment (prakhyatox) [reply src/components/Editor/Organisations/LinkOverlay.vue]: Fixed

File: src/components/Editor/Organisations/LinkOverlay.vue (hunk 3 above)
Comment (prakhyatox) [reply src/components/Editor/Organisations/LinkOverlay.vue]: Fixed

**Pull request merged**